numpy/lib/tests/test_ufunclike.py

```diff
@@ -58,7 +58,9 @@ def __new__(cls, data, metadata=None):
                 return res
 
             def __array_wrap__(self, obj, context=None):
-                if isinstance(obj, MyArray):
+                if not isinstance(obj, MyArray):
+                    obj = obj.view(MyArray)
+                if obj.metadata is None:
```

Comment:
This one is the one change, it now gets a plain array, not a subclass.  The user can actually "detect" that case by checking if `context!=None`